javagen/src/main/java/com/azure/autorest/template/TemplateHelper.java

```diff
@@ -41,6 +41,7 @@ public static String getByteCloneExpression(String propertyName) {
     }
 
     public static void createHttpPipelineMethod(JavaSettings settings, String defaultCredentialScopes, SecurityInfo securityInfo, PipelinePolicyDetails pipelinePolicyDetails, JavaBlock function) {
+        function.line("this.validateBuilder();");
```

Comment:
`validateBuilder()` done in `createHttpPipeline()`.

One reason is that if user provide a `pipeline` instead, validation about "credential" does not need to happen (user could provide an HTTP pipeline that already integrated the credentials into pipeline.

However, the disadvantage here is that the validation on e.g. `endpoint` will also be skipped, if user provides the pipeline.

Reply:
I wonder if we should have two methods for validation:

- `validateClient()` which validates anything used to create the client, but not in the pipeline. Ex, endpoint, service version, etc. Or, maybe this is generated into the client constructor.
- `validatePipeline()` which validates anything used to create the pipeline. Ex, credential information, etc.

Reply:
Emm, yes, that should work.
`validatePipeline()` would be called in `createHttpPipeline()`, `validateClient()` would be called in `createInnerClient()`.

@billwert Let us know your opinion as well.

Reply:
I would go `validateClient()` + `validatePipeline()`, if no further feedback.

Reply:
Change to `validateClient()` alone, after discussion with Srikanta.

The thing in `validatePipeline()` maybe done as
```java
if (this.pipeline == null) {
  // validate credential
}
```
in `validateClient()`.

Reply:
yeah, I think just having `validateClient` is great. (Sorry I missed this ping!)